src/open_producten/producttypen/views.py

```diff
@@ -57,10 +74,48 @@ class VraagViewSet(OrderedModelViewSet):
     serializer_class = VraagSerializer
     lookup_url_kwarg = "id"
     filter_backends = [DjangoFilterBackend]
-    filterset_fields = ["product_type_id", "onderwerp_id"]
+    filterset_fields = ["product_type_id", "thema_id"]
 
 
-class OnderwerpViewSet(OrderedModelViewSet):
-    queryset = Onderwerp.objects.all()
-    serializer_class = OnderwerpSerializer
+class BestandViewSet(OrderedModelViewSet):
+    queryset = Bestand.objects.all()
+    parser_classes = [MultiPartParser]
+    serializer_class = BestandSerializer
     lookup_url_kwarg = "id"
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ["product_type_id"]
+
+
+class ThemaViewSet(OrderedModelViewSet):
+    queryset = Thema.objects.all()
+    serializer_class = ThemaSerializer
+    lookup_url_kwarg = "id"
+
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        errors = []
+        for product_type in ProductType.objects.filter(themas__in=[instance]):
+            if product_type.themas.count() <= 1:
+                errors.append(
+                    _(
+                        "Product Type {} moet aan een minimaal één thema zijn gelinkt."
+                    ).format(product_type)
+                )
+
+        if errors:
+            return Response(
+                data={"product_typen": errors}, status=status.HTTP_400_BAD_REQUEST
+            )
+        try:
+            return super().destroy(request, *args, **kwargs)
+        except ProtectedError:
```

Comment:
Something that could come up in the future is something else being protected other than sub_themas